luaext/lposix.c

```diff
@@ -330,26 +330,64 @@ static int Pmkfifo(lua_State *L)		/** mkfifo(path) */
 }
 
 
+static void set_cloexec(int fd)
+{
+	int flag = fcntl(fd, F_GETFD);
+
+	if (flag == -1 || (flag & FD_CLOEXEC))
+		return;
+
+	fcntl(fd, F_SETFD, FD_CLOEXEC);
+}
+
+static void set_cloexec_all(void)
+{
+	const int min_fd = 3; // don't touch stdin/out/err
+	const int default_open_max = 1024;
+	int fd, open_max;
+
+	open_max = sysconf(_SC_OPEN_MAX);
+	if (open_max == -1) {
+	    open_max = default_open_max;
+	    goto fallback;
+	}
+
+	if (open_max <= default_open_max) {
```

Comment:
I doubt you need to be this conservative. I don't have any measurements to support it, but I'd be very surprised if it is not a win to remove this if statement and always use the code that only traverses the open file descriptors.

Reply:
I wanted to be conservative for the sole reason to minimize the change to current behavior. What you say makes sense though, let me update the patch.